boa/src/builtins/error/syntax.rs

```diff
@@ -12,25 +12,51 @@
 //! [mdn]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/SyntaxError
 
 use crate::{
-    builtins::{function::make_builtin_fn, function::make_constructor_fn},
-    object::ObjectData,
+    builtins::BuiltIn,
+    object::{ConstructorBuilder, ObjectData},
     profiler::BoaProfiler,
+    property::Attribute,
     Context, Result, Value,
 };
 
 /// JavaScript `SyntaxError` impleentation.
 #[derive(Debug, Clone, Copy)]
 pub(crate) struct SyntaxError;
 
-impl SyntaxError {
-    /// The name of the object.
-    pub(crate) const NAME: &'static str = "SyntaxError";
+impl BuiltIn for SyntaxError {
+    const NAME: &'static str = "SyntaxError";
+
+    fn attribute() -> Attribute {
+        Attribute::WRITABLE | Attribute::NON_ENUMERABLE | Attribute::CONFIGURABLE
+    }
+
+    fn init(context: &mut Context) -> (&'static str, Value, Attribute) {
+        let _timer = BoaProfiler::global().start_event(Self::NAME, "init");
 
+        let error_prototype = context.standard_objects().error_object().prototype();
+        let attribute = Attribute::WRITABLE | Attribute::NON_ENUMERABLE | Attribute::CONFIGURABLE;
+        let range_error_object = ConstructorBuilder::with_standard_object(
```

Comment:
`range_error_object` should be `syntax_error_object`, like in `type.rs`

Reply:
same here